File: week4/app/src/main/java/com/example/fragmentjava/ViewFragment.java
package com.example.fragmentjava;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import com.example.fragmentjava.databinding.ViewFragmentBinding;

// This is the class declaration for the ViewFragment class that extends Fragment
public class ViewFragment extends Fragment {

    // Declaration of the binding variable with ViewFragmentBinding data type
    private ViewFragmentBinding binding;

    // Constructor for the ViewFragment class
    public ViewFragment(){}

    // onCreateView() method, called when the view is created
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        // Inflate the View for this fragment using the binding
        binding = ViewFragmentBinding.inflate(inflater, container, false);
        View view = binding.getRoot();

        // Read the text message from SharedPreferences
        // Open the shared preferences for this app's named "Message" file.
        // This file is private and can't be accessed by other apps.
        SharedPreferences sharedPref= requireActivity().
                getSharedPreferences("Message", Context.MODE_PRIVATE);
        // Retrieve the string value associated with the key "message" from
        // the shared preferences file. If the value is not found or has been
        // removed, the default value is null.
        String message = sharedPref.getString("message", null);

        // Set text in textView to the retrieved message
        binding.textMessage.setText("Message from AddFragment: " + message);

        // Return the View object
        return view;
    }

    // onDestroyView() method, called when the fragment's view is destroyed
    @Override
    public void onDestroyView() {
        super.onDestroyView();
        // Set binding variable to null to avoid memory leaks
        binding = null;
    }
}

File: week4/FragmentKotlin/app/src/main/java/com/example/fragmentkotlin/ViewFragment.kt
package com.example.fragmentkotlin

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.fragmentkotlin.databinding.ViewFragmentBinding

// display a message stored in SharedPreferences in a TextView.
class ViewFragment: Fragment() {
    // Declare a private nullable variable to hold the ViewBinding.
    private var _binding: ViewFragmentBinding? = null

    // Create a binding property that returns the binding or throws an exception if it's null.
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the ViewFragmentBinding and assign it to _binding.
        _binding = ViewFragmentBinding.inflate(inflater, container, false)

        // Get the SharedPreferences instance for the "Message" key.
        val sharedPref: SharedPreferences = requireActivity().getSharedPreferences(
            "Message",
            Context.MODE_PRIVATE
        )

        // Get the value of the "message" key from SharedPreferences.
        val message = sharedPref.getString("message", null)

        // Set the text of the textMessage TextView to display the message from SharedPreferences.
        binding.textMessage.text = "Message from AddFragment: $message"

        // Return the root view of the binding.
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()

        // Set the _binding variable to null to release the ViewBinding.
        _binding = null
    }
}
